optuna/storages/_rdb/models.py

```diff
@@ -447,23 +449,85 @@ def where_trial_id(cls, trial_id: int, session: orm.Session) -> List["TrialValue
     trial_id = Column(Integer, ForeignKey("trials.trial_id"), nullable=False)
     step = Column(Integer, nullable=False)
     intermediate_value = Column(Float(precision=FLOAT_PRECISION), nullable=True)
-    intermediate_value_type = Column(Enum(FloatTypeEnum), nullable=False)
+    intermediate_value_type = Column(Enum(TrialIntermediateValueType), nullable=False)
 
     trial = orm.relationship(
         TrialModel, backref=orm.backref("intermediate_values", cascade="all, delete-orphan")
     )
 
+    @classmethod
+    def _intermediate_value_to_stored_repr(
+        cls,
+        value: float,
+    ) -> Tuple[Optional[float], TrialIntermediateValueType]:
+        if math.isnan(value):
+            return (None, cls.TrialIntermediateValueType.NAN)
+        elif value == float("inf"):
+            return (None, cls.TrialIntermediateValueType.INF_POS)
+        elif value == float("-inf"):
+            return (None, cls.TrialIntermediateValueType.INF_NEG)
+        else:
+            return (value, cls.TrialIntermediateValueType.FINITE)
+
+    @classmethod
+    def _stored_repr_to_intermediate_value(
+        cls, value: Optional[float], float_type: TrialIntermediateValueType
+    ) -> float:
+        if float_type == cls.TrialIntermediateValueType.NAN:
+            assert value is None
+            return float("nan")
+        elif float_type == cls.TrialIntermediateValueType.INF_POS:
+            assert value is None
+            return float("inf")
+        elif float_type == cls.TrialIntermediateValueType.INF_NEG:
+            assert value is None
+            return float("-inf")
+        else:
+            assert float_type == cls.TrialIntermediateValueType.FINITE
+            assert value is not None
+            return value
+
+    @classmethod
+    def make_record_with_original_intermediate_value(
+        cls, *, original_intermediate_value: float, **kwargs: Any
+    ) -> "TrialIntermediateValueModel":
+        assert "intermediate_value" not in kwargs
+        assert "intermediate_value_type" not in kwargs
+        (
+            stored_intermediate_value,
+            intermediate_value_type,
+        ) = cls._intermediate_value_to_stored_repr(original_intermediate_value)
+        return cls(
+            **kwargs,
+            intermediate_value=stored_intermediate_value,
+            intermediate_value_type=intermediate_value_type,
+        )
+
+    def get_original_intermediate_value(self) -> float:
+        return self._stored_repr_to_intermediate_value(
+            self.intermediate_value, self.intermediate_value_type
+        )
+
+    def set_original_intermediate_value(self, value: float) -> None:
+        (
+            stored_intermediate_value,
+            intermediate_value_type,
+        ) = self._intermediate_value_to_stored_repr(value)
+        self.intermediate_value = stored_intermediate_value
+        self.intermediate_value_type = intermediate_value_type
```

Comment:
I think it is more easy to understand to call `_intermediate_value_to_stored_repr` and `_stored_repr_to_intermediate_value` directly in `storage.py`, so we should remove these helper functions. What do you think? @contramundum53 @toshihikoyanase @c-bata

Reply:
That makes sense. I'll fix it.